tests/test_requests.py

```diff
@@ -698,3 +777,35 @@ def test_request_form_invalid_content_type(app):
     request, response = app.test_client.post("/", json={"test": "OK"})
 
     assert request.form == {}
+
+
+def test_url_for_with_forwarded_request(app):
+    @app.get("/")
+    def handler(request):
+        return text("OK")
+
+    @app.get("/another_view/")
+    def view_name(request):
+        return text("OK")
+
+    request, response = app.test_client.get("/", headers={
+        'X-Forwarded-Proto': 'https',
+    })
+    assert app.url_for('view_name') == '/another_view'
+    assert app.url_for('view_name', _external=True) == 'http:///another_view'
```

Comment:
That result is fishy. `http:///` ?

Reply:
That's the missing UT case for `Sanic.url_for`. But there's no good way to fix it since getting a valid hostname needs `Request` context as in this PR.

Reply:
You can provide some `kwargs` to `url_for`, such as `_server` that can have your valid hostname embed into, like `_server=request.headers.get("Host")` or `_scheme`, like `_scheme=request.headers.get("X-Forwarded-Proto")`. You can check most of them [here](https://github.com/huge-success/sanic/blob/2af229eb1ad89e4d064196d98b3f31630c0423d3/sanic/app.py#L777-L785). I think we should properly document those `kwargs` from the `url_for` method.